src/v/coproc/sys_refs.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/partition_manager.h"
+#include "storage/api.h"
```

Comment:
Only forward declarations are required here.

I would tend to put them in `storage/fwd.h` and `cluster/fwd.h`